test/currencyConvertor.ts

```diff
@@ -0,0 +1,78 @@
+import { ethers, waffle } from 'hardhat'
+import chai from "chai";
+import { solidity } from "ethereum-waffle";
+import CurrencyConvertorArtifact from '../artifacts/contracts/proxies/CurrencyConvertor.sol/CurrencyConvertor.json';
+import ZeroExExchangeWrapperArtifact from '../artifacts/contracts/exchange-wrappers/ZeroExExchangeWrapper.sol/ZeroExExchangeWrapper.json';
+import StarkwareArtifact from '../artifacts/contracts/exchange-wrappers/Starkware.sol/Starkware.json';
+import UsdcContractArtifact from '../artifacts/contracts/tokens/mockUsdcContract.sol/MockUsdcContract.json';
+
+const { deployContract } = waffle
+
+
+chai.use(solidity);
+const { expect } = chai;
+describe("CurrencyConvertor", () => {
+  let currencyConvertor: any;
+  let zeroExExchangeWrapper: any;
+  let starkware: any;
+  let usdc: any;
+  beforeEach(async () => {
+    const signers = await ethers.getSigners();
+
+    zeroExExchangeWrapper = await deployContract(
+      signers[0],
+      ZeroExExchangeWrapperArtifact,
+    );
+    starkware = await deployContract(
+      signers[0],
+      StarkwareArtifact,
+    );
+    usdc = await deployContract(
+      signers[0],
+      UsdcContractArtifact,
+    )
+
+    await usdc.mint(
+      zeroExExchangeWrapper.address,
+      1000e6,
+    )
+    await usdc.mint(
+      starkware.address,
+      1000e6,
+    )
+
+    currencyConvertor = await deployContract(
+      signers[0],
+      CurrencyConvertorArtifact,
+      [
+        starkware.address,
+        usdc.address, // placeholder for USDC
+        156,
+      ],
+    );
+
+    await usdc.mint(
+      currencyConvertor.address,
+      1000e6,
+    )
```

Comment:
It's possible your tests are still failing because you didn't mint any USDC to `signers[0]`. You can verify how much USDC they have by logging the following function call:

`console.log(await usdc.balanceOf(await signers[0].getAddress()));`

It seems like you've minted USDC to the multiple contract addresses but not the signer itself, so you're probably transferring USDC that the signer does not have.

Reply:
nice this combined with approving for the proxy allowed me to transfer to the exchange but I cant get money back out of the exchange haha